geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/AttributeIdxStrategy.scala

```diff
@@ -467,41 +482,60 @@ object AttributeIndexStrategy extends StrategyProvider {
         }
         (prop, AccRange.prefix(getEncodedAttrIdxRow(sft, prop, value)))
 
+      case n: Not =>
+        val f = n.getFilter.asInstanceOf[PropertyIsNull] // this should have been verified in getStrategy
+        val prop = f.getExpression.asInstanceOf[PropertyName].getPropertyName
+        (prop, allRange(sft, prop))
+
       case _ =>
         val msg = s"Unhandled filter type in attribute strategy: ${filter.getClass.getName}"
         throw new RuntimeException(msg)
     }
 
   private def greaterThanRange(sft: SimpleFeatureType, prop: String, lit: AnyRef): AccRange = {
-    val rowIdPrefix = getTableSharingPrefix(sft)
     val start = new Text(getEncodedAttrIdxRow(sft, prop, lit))
-    val endPrefix = AttributeTable.getAttributeIndexRowPrefix(rowIdPrefix, sft.getDescriptor(prop))
-    val end = AccRange.followingPrefix(new Text(endPrefix))
+    val end = upperBound(sft, prop)
     new AccRange(start, false, end, false)
```

Comment:
not really actually... I think the old way nulls would be counted as < everything, so the ranges haven't changed. I just extracted out the common code here and in the big match above into methods.